src/cargo/util/config/target.rs

```diff
@@ -67,19 +67,13 @@ pub(super) fn load_target_cfgs(config: &Config) -> CargoResult<Vec<(String, Targ
 
 /// Returns true if the `[target]` table should be applied to host targets.
 pub(super) fn get_target_applies_to_host(config: &Config) -> CargoResult<bool> {
-    if config.cli_unstable().target_applies_to_host {
-        if let Ok(target_applies_to_host) = config.get::<bool>("target-applies-to-host") {
-            Ok(target_applies_to_host)
-        } else {
-            Ok(!config.cli_unstable().host_config)
-        }
-    } else if config.cli_unstable().host_config {
+    let enabled = config.cli_unstable().target_applies_to_host;
+    if enabled && config.cli_unstable().host_config {
         anyhow::bail!(
-            "the -Zhost-config flag requires the -Ztarget-applies-to-host flag to be set"
+            "the -Zhost-config flag requires the `target-applies-to-host` option to be disabled"
```

Comment:
This message doesn't appear to describe the requirement accurately, as `target-applies-to-host` is currently both a nightly feature and a boolean config option(separate originally but combined in this PR), either way disabled implies that the nightly feature be disabled(ie unset entirely), while in reality this actually appears to require a particular boolean config value to be set(ie `-Ztarget-applies-to-host=no`/`-Ztarget-applies-to-host=false`).